File: mobile/lib/features/invoices/domain/currency.dart
/// A currency an invoice may be issued in.
///
/// [minorUnits] is the number of decimal places the currency is quoted in — 2 for ILS, USD, EUR and
/// GBP, 3 for JOD. It drives every rounding decision in the on-device totals preview, so the app
/// rounds exactly where the server does rather than assuming two decimals everywhere.
class Currency {
  const Currency({
    required this.code,
    required this.name,
    required this.symbol,
    required this.minorUnits,
    required this.active,
  });

  final String code;
  final String name;
  final String symbol;
  final int minorUnits;
  final bool active;

  factory Currency.fromJson(Map<String, dynamic> json) {
    return Currency(
      code: json['code'] as String,
      name: json['name'] as String,
      symbol: json['symbol'] as String,
      minorUnits: json['minorUnits'] as int,
      active: json['active'] as bool? ?? true,
    );
  }
}

File: backend/src/main/java/com/bsharat/triosuite/catalog/Currency.java
package com.bsharat.triosuite.catalog;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

/**
 * An ISO 4217 currency.
 *
 * <p>{@code minorUnits} is the number of decimal places the currency is quoted in and drives every
 * rounding decision in {@link com.bsharat.triosuite.invoice.InvoiceCalculator}: 2 for ILS, USD, EUR
 * and GBP, 3 for JOD.
 */
@Entity
@Getter
@Setter
@Table(name = "currencies")
public class Currency {

    @Id
    @Column(length = 3, nullable = false)
    private String code;

    @Column(nullable = false, length = 60)
    private String name;

    @Column(nullable = false, length = 8)
    private String symbol;

    @Column(name = "minor_units", nullable = false)
    private int minorUnits;

    @Column(name = "is_active", nullable = false)
    private boolean active = true;
}
